genie-web/src/main/java/com/netflix/genie/web/services/impl/AgentFilterServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.web.services.impl;
+
+import com.netflix.genie.common.internal.dto.v4.AgentClientMetadata;
+import com.netflix.genie.web.services.AgentFilterService;
+import com.netflix.genie.web.util.AgentMetadataInspector;
+import com.netflix.genie.web.util.InspectionReport;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.validation.Valid;
+import java.util.List;
+
+/**
+ * Implementation of {@link AgentFilterService} which delegates iterates through an ordered list of
+ * {@link AgentMetadataInspector}. Giving them the chance to accept and reject agents based on their provided
+ * metadata. This filter accepts by default if none of the inspectors rejected.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Slf4j
+public class AgentFilterServiceImpl implements AgentFilterService {
+
+    private final List<AgentMetadataInspector> agentMetadataInspectorList;
+
+    /**
+     * Constructor.
+     *
+     * @param agentMetadataInspectorList the list of inspectors to consult
+     */
+    public AgentFilterServiceImpl(
+        final List<AgentMetadataInspector> agentMetadataInspectorList
+    ) {
+        this.agentMetadataInspectorList = agentMetadataInspectorList;
+    }
+
+    @Override
+    public InspectionReport inspectAgentMetadata(@Valid final AgentClientMetadata agentClientMetadata) {
+        for (final AgentMetadataInspector agentMetadataInspector : agentMetadataInspectorList) {
+            final InspectionReport inspectionReport = agentMetadataInspector.inspect(agentClientMetadata);
+            final InspectionReport.Decision decision = inspectionReport.getDecision();
+            final String message = inspectionReport.getMessage();
+
+            log.debug(
+                "Inspector: {} inspected: {}, decision: {} ({})",
+                agentMetadataInspector,
+                agentClientMetadata,
+                decision.name(),
+                message
+            );
+
+            if (decision == InspectionReport.Decision.REJECT) {
+                return InspectionReport.newRejection("Rejected by: " + agentMetadataInspector + ": " + message);
```

Comment:
You're making an assumption as to the format of the `toString` for `agentMetadataInspector` here. Would it be better to use the class name? The debug statement as well. Either way you should probably assign the value you want to use once instead of potentially implicitly calling `toString` twice